hamilton/driver.py

```diff
@@ -638,20 +638,80 @@ def raw_execute(
         return out
 
     @capture_function_usage
-    def list_available_variables(self) -> List[Variable]:
+    def list_available_variables(
```

Comment:
Feels like this should be a `query_available_variables` function, although it could work either way

Reply:
@kzk2000 thoughts on a new or old function? 

I'm more inclined to have one function.